core/exception_vectors.S

```diff
@@ -491,7 +530,15 @@ NMIExceptionHandler:
         .balign  4
 UserExceptionHandler:
         .type   UserExceptionHandler, @function
-        xsr     a0, excsave1    # a0 now contains sp
+        // save a0, a1 to debug_saved_ctx before stack pointer is affected
```

Comment:
if the "break 1" instruction moves from  LUserFailOtherExceptionCause to UserExceptionVector (which already examines exccause for LOADSTORE, and probably has enough space to check for CAUSE_LVL1INT and break if it's not hat), is it then possible to trigger the debugger before we save interrupt state, and eliminate the need to save these registers twice?